### PR TITLE
[16.0][IMP] report_xml: Added `xml_extension`

### PR DESCRIPTION
**Pull request by Mantux11**
Adds a way to choose the file extension for your XML file, by adding the flexibility of appending more choices. For example, instead of getting `.xml`, this allows to have the report download as `.svg` or even good old `.html` if needed. This is also very useful for some localization purposes, where some scarce software uses uncommon file extensions for their XML files, like `.ffdata` for specific accounting reports here in Lithuania.

The change is not breaking, as we set the default to be `.xml` like it was by default anyway.

### CODE REVIEW

**Review by gaikaz — APPROVED** [review]

**Comment by gaikaz** [comment]
@OCA/reporting-engine-maintainers 
This is coming from Via laurea, where we have an extension module exactly for this.
We now decided to suggest it to the main module here on OCA, as we see it as a useful feature for the base module.

Could we get opinions and/or more reviews?

File: report_xml/__manifest__.py
Comment (pedrobaeza) [inline report_xml/__manifest__.py]: Please don't change manually the version. It will be done by bot on merge.

File: report_xml/models/ir_actions_report.py
Comment (pedrobaeza) [inline report_xml/models/ir_actions_report.py]: Do it a Char field for not having to program an extension module for further ones.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Review by lmignon — CHANGES_REQUESTED** [review]
Thank you @Mantux11 for your contribution. Your proposal makes sense. Can you take into account the Pedro's comments PLZ?

**Comment by gaikaz** [comment]
I didn't expect to get that quick of a response. Thank you 🙏🏻 
Changing the field from Selection to Char is a good point. It also makes it a lot more testable. Therefore, we'll also add a test for this new feature. 
@Mantux11 is currently having a time-off, but should come back to this on Monday.

**Comment by HviorForgeFlow** [comment]
@Mantux11 pre-commit will go green after merging this and rebasing on your side:
https://github.com/OCA/reporting-engine/pull/724

CC @pedrobaeza

**Comment by pedrobaeza** [comment]
You can rebase now.

**Comment by Mantux11** [comment]
@HviorForgeFlow It's not because of `pre-commit`. I'm trying to write a test function for this implentation. and locally I couldn't truly test everything. So i'm pushing here, to see if response from requests is getting correct, and I could equal if extension is correct.

**Comment by Mantux11** [comment]
@HviorForgeFlow I will do that later. Now I'm trying to write a test for my added functionality, locally I couldn't test everything.

**Comment by Mantux11** [comment]
@pedrobaeza @lmignon I tried to request a re-review from both, but idk why if I request from one, then from other person request is removed :D I included test for this functionality. And also rebase is done.

File: report_xml/models/ir_actions_report.py
Comment (pedrobaeza) [inline report_xml/models/ir_actions_report.py]:
```suggestion
        help="Extension for XML Reports, by default is `xml`",
```

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Please avoid hammering the OCA CI doing a lot of pushes, and instead, test on local, and also squash your commits into one.

**Comment by Mantux11** [comment]
@pedrobaeza Sorry, that's the last push for all test code refactoring, all tests is successful as I see. Squash would be the last commit push here.

**Review by gaikaz — APPROVED** [review]

**Comment by Mantux11** [comment]
@lmignon Could you review all changes?

**Review by DonatasV — APPROVED** [review]

**Comment by Mantux11** [comment]
@OCA/reporting-engine-maintainers Could anyone approve and merge it? Thanks:)

**Comment by pedrobaeza** [comment]
As my comments were attended, which were the @lmignon's changes requested, I'm merging:

/ocabot merge minor

Thanks for the feature.

**Pull request merged**
